Src/Support/Google.Apis.Auth/OAuth2/UserCredential.cs

```diff
@@ -31,11 +33,19 @@ namespace Google.Apis.Auth.OAuth2
     /// OAuth 2.0 credential for accessing protected resources using an access token, as well as optionally refreshing 
     /// the access token when it expires using a refresh token.
     /// </summary>
-    public class UserCredential : ICredential, IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler
+    public class UserCredential : ICredential, IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler, ITokenAccessWithExtraInformation
     {
+        private const string QuotaProjectIdKey = "x-goog-user-project";
```

Comment:
This isn't just the key in the dictionary - it's used as the header, right? I think we need to make the correspondence between header names and dictionary keys clearer.